src/librustdoc/doctest/make.rs

```diff
@@ -429,56 +410,40 @@ impl DocTestBuilder {
     }
 }
 
-fn reset_error_count(psess: &ParseSess) {
-    // Reset errors so that they won't be reported as compiler bugs when dropping the
-    // dcx. Any errors in the tests will be reported when the test file is compiled,
-    // Note that we still need to cancel the errors above otherwise `Diag` will panic on
-    // drop.
-    psess.dcx().reset_err_count();
-}
-
-const DOCTEST_CODE_WRAPPER: &str = "fn f(){";
-
 fn parse_source(
     source: &str,
     crate_name: &Option<&str>,
     parent_dcx: Option<DiagCtxtHandle<'_>>,
     span: Span,
 ) -> Result<ParseSourceInfo, ()> {
-    use rustc_errors::DiagCtxt;
-    use rustc_errors::emitter::{Emitter, HumanEmitter};
-    use rustc_span::source_map::FilePathMapping;
-
     let mut info =
         ParseSourceInfo { already_has_extern_crate: crate_name.is_none(), ..Default::default() };
 
+    const DOCTEST_CODE_WRAPPER: &str = "fn f(){";
     let wrapped_source = format!("{DOCTEST_CODE_WRAPPER}{source}\n}}");
 
-    let filename = FileName::anon_source_code(&wrapped_source);
+    // Any parse errors should also appear when the doctest is compiled for real,
+    // so just suppress them here.
```

Comment:
Also for e.g., `compile_fail` we don't necessary want to output any errors "unless e.g., `--nocapture` or `--test-args --show-output`" …; so it really isn't our responsibility.

Therefore elaborate that source comment.